### ISSUE TITLE
USBs spawning without software

### ISSUE THREAD

**Issue by heebejeebes** [CleverRaven/Cataclysm-DDA#41410]
# Describe the bug

None of the USB drives in the game are spawning with any software attached. Spent hours traveling to the most likely locations (labs and regional dumps) and found no Medisoft or HackPROs. USBs are indeed spawning but with nothing in them.

# Steps To Reproduce

1. Generate a new world.
2. Travel to each lab, or dump
3. Check each site for USB drives with attached software instead of empty drives

# Expected behavior

When traveling to one of these locations you should be able to find some drives with software attached relatively commonly.

# Screenshots

Not applicable

# Versions and configuration

 - OS: Windows 10 Professional x64
 - Game Version: 0.E-3303-ge6b1dea
 - Graphics version: Tiles
 - Ingame language: English
 - Mods loaded: Alternative Map Key, DinoMod
 -->

# Additional context

Suspect this is a nested container bug that fell through the cracks. In the debug menu I spawned a HackPRO and it appeared in my inventory without a USB. This leads me to believe that software instances behave like liquids do and need a valid container to exist inside.

A possible workaround might have been to allow us to use [f] in the spawn item menu to spawn contained software in a USB. Currently, however, spawning contained software does nothing different than spawning uncontained software. This makes building control laptops currently impossible unless you spawn a finished item.

Crash: No crashes related to this bug

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/cant-find-any-usbs-with-software/23970/4

**Comment by kevingranade** [comment]
I vaguely remember this being addressed, need to verify.

**Comment by PatrikLundell** [comment]
I've been able to find USBs with SW (HackPRO, medical, and mixed, I think) using 0.E2 stable, using the "mods" that were standard at that time (DDA and no NPC needs). That might support the recollection that it has been addressed at some time.

**Comment by Cyprex** [comment]
On 0.E.3 I've found only mathemax in a doctors office, though I only looted a small starting city.
In a different world on an experimental release from a few days ago I've also only found mathemax. I've looted:
various houses
a doctors office and a veterinarian clinic
3 electronics stores
Total haul:
![Screenshot_20210110_190522-1](https://user-images.githubusercontent.com/35031555/104131642-eef24300-5377-11eb-85cc-b8f40cbd1413.png)
Note that some of the SD cards were wiped by using them in an e-ink tablet.
Game report for said experimental world: (The world was generated with this release and only been played on it.)
- OS: Linux
    - OS Version: Distributor ID: Debian; Description: Debian GNU/Linux 10 (buster); Release: 10; Codename: buster; 
- Game Version: 0.E-8687-ged7efb8 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]
